### PR TITLE
Split cuda utils.h & fix includes

### PR DESCRIPTION
**Pull request by mzient**
## Category:
**Refactoring** (*Redesign of existing code that doesn't affect functionality*)

## Description:
The file cuda_utils.h mixed concerns. It's now split into cuda_util.h and cuda_rt_utils.h
cuda_utils.h was include excessively instead of other, more appropriate headers. This is now fixed.

### Affected modules and functionalities:
No functionality is affected.

### Tests:
This is just code movement, tests should not be affected and aren't needed.
- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [X] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [X] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [X] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Review by JanuszL — APPROVED** [review]

File: include/dali/core/cuda_rt_utils.h [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  }
+  static int dev_count = []() {
+    int ndevs = 0;
+    CUDA_CALL(cudaGetDeviceCount(&ndevs));
+    return ndevs;
+  }();
+  static std::vector<bool> read(dev_count, false);
+  static std::vector<cudaDeviceProp> properties(dev_count);
+  if (!read[device_id]) {
+    CUDA_CALL(cudaGetDeviceProperties(&properties[device_id], 0));
```
Comment (JanuszL) [inline include/dali/core/cuda_rt_utils.h:59]:
```suggestion
    CUDA_CALL(cudaGetDeviceProperties(&properties[device_id], device_id));
```

Comment (mzient) [reply include/dali/core/cuda_rt_utils.h:59]: Nice catch! It was lurking there for quite a time.

Comment (JanuszL) [reply include/dali/core/cuda_rt_utils.h]: Yes, it was...

File: dali/kernels/imgproc/roi.h [hunk 2]
```diff
@@ -193,8 +194,8 @@ Roi<spatial_dims> AdjustRoi(const Roi<spatial_dims> *roi, const TensorShape <ndi
 template <int ndims, int spatial_dims = ndims - 1>
 std::vector<Roi<spatial_dims>>
 AdjustRoi(span<const Roi<spatial_dims>> rois, const TensorListShape <ndims> &shapes) {
-  DALI_ENFORCE(rois.empty() || rois.size() == shapes.num_samples(),
-               "Either provide `rois` for every corresponding `shape`, or none.");
+  if (rois.empty() || rois.size() == shapes.num_samples())
+    throw std::invalid_argument("Either provide `rois` for every corresponding `shape`, or none.");
```
Comment (jantonguirao) [inline dali/kernels/imgproc/roi.h:198]: This is not correct. You changed the condition for the error.

Comment (jantonguirao) [reply dali/kernels/imgproc/roi.h]: Fixed

**Comment by jantonguirao** [comment]
!build

**Review by jantonguirao — APPROVED** [review]

**Pull request merged**